src/server/protocol_client.cc

```diff
@@ -56,17 +56,23 @@ static ProtocolClient::SSL_CTX* CreateSslClientCntx() {
   const auto& tls_key_file = GetFlag(FLAGS_tls_key_file);
   unsigned mask = SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT;
 
-  CHECK_EQ(1, SSL_CTX_use_PrivateKey_file(ctx, tls_key_file.c_str(), SSL_FILETYPE_PEM));
-  const auto& tls_cert_file = GetFlag(FLAGS_tls_cert_file);
+  // Load client certificate if given.
+  if (!tls_key_file.empty()) {
```

Comment:
I  am not quite sure what this check adds. From my understanding you want to allow clients(replicas) configured without a private/public key. What's the point of having TLS then? Would this even work?

Reply:
Yeah, it works (you're correct that I need to add some tests though).
In this configuration the TLS layer provides encryption and server authentication with the server certificate, and the password authenticates the client.

Reply:
Ohhh I see 👍